daemon/internal/cloud_client/long_poll_service_event.go

```diff
@@ -0,0 +1,43 @@
+package cloud_client
+
+import (
+	"time"
+
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/util"
+	"github.com/akitasoftware/akita-libs/akid"
+)
+
+// Long-polls the cloud for updates to a service.
+type longPollServiceEvent struct {
+	serviceID akid.ServiceID
+}
+
+func newLongPollServiceEvent(serviceID akid.ServiceID) longPollServiceEvent {
+	return longPollServiceEvent{
+		serviceID: serviceID,
+	}
+}
+
+func (event longPollServiceEvent) handle(client *cloudClient) {
+	printer.Debugf("Polling for changed traces at service %s\n", akid.String(event.serviceID))
+	currentTraces := client.getCurrentTraces(event.serviceID)
+	frontClient := client.frontClient
+	go func() {
+		// Send a request to the cloud containing a list of the traces currently
+		// being logged. The response will be a list of new traces to log and a
+		// list of traces that should finish logging.
+		activeTraceDiff, err := util.LongPollActiveTracesForService(frontClient, event.serviceID, currentTraces)
+		if err != nil {
+			// Log the error, wait a bit, and try again.
+			printer.Debugf("Error while polling service ID %s: %v\n", akid.String(event.serviceID), err)
+			time.Sleep(LONG_POLL_INTERVAL)
+			client.eventChannel <- event
+			return
+		}
+
+		// Enqueue a changeActiveTracesEvent for the main goroutine to handle.
+		printer.Debugf("Enqueuing changed-traces event for service %s\n", akid.String(event.serviceID))
+		client.eventChannel <- newChangeActiveTracesEvent(event.serviceID, activeTraceDiff)
```

Comment:
I find it kind of subtle that newChangeActiveTracesEvent is responsible for re-enqueing the long poll.  Is there a reason not to do it immediately?

Reply:
We don't always want to resume long polling—the handler for `changeActiveTracesEvent` sometimes returns without re-enqueuing the long poll. I'll add comments to clarify where the responsibility lies.

Reply:
Done in 1190f53.